internal/stage7.go

```
@@ -13,12 +14,30 @@ func testType2(stageHarness *test_case_harness.TestCaseHarness) error {
+	// The other one is in a directory which comes after the first one's directory on PATH
+	// The other one should still be found & used
+	executable_expected_to_be_not_found := "my_exe"
 	executableDir, err := SetUpCustomCommands(stageHarness, shell, []CommandDetails{
+		{CommandType: "signature_printer", CommandName: executable_expected_to_be_not_found, CommandMetadata: getRandomString()},
+	}, true)
+	if err != nil {
+		return err
+	}
+	notExePath := filepath.Join(executableDir, executable_expected_to_be_not_found)
+	currentPerms, _ := os.Stat(notExePath)
+	os.Chmod(notExePath, currentPerms.Mode() & ^os.FileMode(0o111))
```

golangci-lint (1.62.2)

Error return value of `os.Chmod` is not checked

(errcheck)